### PR TITLE
Added hot plug support for on-going traffic case. Before removal we a…

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
This pull request has the following changes:

- Before performing hot removal we are shutdown the userpf.  This will prevent system crash while traffic is going on.
- xbmgmt will poll till shutdown finishes and then try to remove userpf and mgmt pf 
- If we are running xbmgmt on VM environment then it will print a message and gracefully exits.

### CODE REVIEW

**Comment by chvamshi-xilinx** [comment]
ok to test

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp [hunk 1]
```diff
@@ -114,6 +134,46 @@ int hotplugHandler(int argc, char *argv[])
...
+{
+    std::string errmsg;
+    int shutdownStatus = -EINVAL;
+    int wait  = 0;
+
+    /* "echo 1 > /sys/bus/pci/<EndPoint>/shutdown" to trigger shutdown of the device */
+    std::string path = dev->get_sysfs_path("", "shutdown");
+    if (!boost::filesystem::exists(path)) 
+    {
+        return -EEXIST;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: shouldn't you return ENOENT?

Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: Thanks for the comments.  I will make the changes and publish them again.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp [hunk 2]
```diff
@@ -29,9 +29,11 @@
 #define SYSFS_PATH      "/sys/bus/pci/devices"
 #define XILINX_VENDOR   "0x10ee"
 #define XILINX_US       "0x9134"
+#define POLL_TIMEOUT    60      /* Set a pool timeout as 60sec */
 
 static int hotplugRescan(void);
 static int removeDevice(const std::shared_ptr<pcidev::pci_device> dev);
+static int shutdownDevice(const std::shared_ptr<pcidev::pci_device> dev);
```
Comment (stsoe) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: change to 'const pcidev::pci_device* dev' and call as 'shutdownDevice(uDev.get())'
Ditto for removeDevice.

Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: I have updated the code. But it is a bit different way. Passed device index as a parameter. The same implementation is followed in other files. Please review it once.

**Comment by chvamshi-xilinx** [comment]
retest this please

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp [hunk 3]
```diff
@@ -29,9 +29,11 @@
 #define SYSFS_PATH      "/sys/bus/pci/devices"
 #define XILINX_VENDOR   "0x10ee"
 #define XILINX_US       "0x9134"
+#define POLL_TIMEOUT    60      /* Set a pool timeout as 60sec */
 
 static int hotplugRescan(void);
-static int removeDevice(const std::shared_ptr<pcidev::pci_device> dev);
+static int removeDevice(unsigned index, bool is_userpf);
```
Comment (stsoe) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: Please specify type.     unsigned what?

**Pull request merged**